gen.py

```diff
@@ -104,25 +133,69 @@ def schema_extra(schema: Dict[str, Any], _) -> None:
             schema["maxProperties"] = 1
 
 
+class VarPath(BaseModel):
+    __root__: str = Field(
+            ..., description="Path to params file for parametrization."
+    )
+
+
 class VarDecl(BaseModel):
     # {"foo" (str) : "foobar" (Any) }
-    __root__: Dict[VarKey, Any]
+    __root__: Dict[VarKey, Any] = Field(
+            ..., description="Mapping for parametrization."
+    )
 
 
 class Vars(BaseModel):
-    __root__: List[Union[FilePath, VarDecl]]
+    __root__: List[Union[VarPath, VarDecl]]
+
+
+STAGE_VARS_DESC = """\
+Sequence of variables for parametrization.
+
+May include any mapping or a path to a params file (which may be a string or a \
+mapping to params in the file).
+
+Use in the stage with the `${}` substitution expression."""
+
+CMD_DESC = """\
+(Required) Command to run (anything your system terminal can run).
+
+Can be a string or a sequence of commands."""
+
+PARAMS_DESC = """\
+Sequence of dot-separated parameter dependency keys to track from `params.yaml`.
 
+May contain other YAML/JSON/TOML/Python parameter file names, with a \
+sub-sequence of the param names to track in them (leave empty to include all).\
+"""
+
+METRICS_DESC = "Sequence of metrics of the stage written to JSON/TOML/YAML for \
+evaluation."
```

Comment:
Why sequence?
Are metrics always for evaluation? Maybe we don't need to state that just to avoid sounding opinionated (even if it's embedded in the concept of "metrics").